zebra-state/src/service/non_finalized_state.rs

```diff
@@ -70,6 +73,15 @@ impl NonFinalizedState {
     /// Commit block to the non-finalized state.
     pub fn commit_block(&mut self, prepared: PreparedBlock) {
         let parent_hash = prepared.block.header.previous_block_hash;
+        let (height, hash) = (prepared.height, prepared.hash);
+
+        let canopy_height = Canopy.activation_height(self.network).unwrap();
+        if height < canopy_height {
+            panic!(
+                "commit_block can only be applied to block heights that are equal \
+                or greater than canopy network upgrade height"
+            );
```

Comment:
Looks good, just tweaking some of the wording of the panic message and variables:
```suggestion
        let canopy_activation_height = Canopy.activation_height(self.network).unwrap();
        if height < canopy_activation_height {
            panic!(
                "invalid non-finalized block height: the canopy checkpoint is mandatory, pre-canopy blocks must be committed to the state as finalized blocks"
            );
```